src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const obj = { ...state };
+  const arr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    for (const key in actions[i]) {
+      if (actions[i][key] === 'removeProperties') {
+        for (let j = 0; j < actions[i].keysToRemove.length; j++) {
+          delete obj[actions[i].keysToRemove[j]];
+        }
+        arr[i] = { ...obj };
```

Comment:
4. [DONT REPEAT YOURSELF] - If you perform same action in all `switch` cases - do it just once afterwards.